heron/instance/src/java/org/apache/heron/network/StreamManagerClient.java

```diff
@@ -286,6 +289,17 @@ private void handleStartStatefulRequest(
     inControlQueue.offer(instanceControlMsg);
   }
 
+  private void handleCheckpointSaved(
+      CheckpointManager.StatefulConsistentCheckpointSaved message) {
+    LOG.info("Received a StatefulCheckpointSaved message:with checkpoint id: "
```

Comment:
remove the first `:`

Reply:
> It seems the message loop is messing the later half part of sending ack from instance to tmaster to confirm the commit for a checkpoint id.

You mean "missing" right? We decided that we don't need to send `ack` back to TMaster. This is because since we will block executing tuples (or producing tuples for spouts) if the `postSave` callback on the task instance has not yet exited. As a result, the next checkpoint will never become globally consistent (since the blocked task will not have the chance to persist its state for the next checkpoint).

Therefore, all other tasks in the topology will be blocked on their `postSave` for the next checkpoint, which is the desired behaviour :)